packages/firebase_remote_config/lib/src/remote_config.dart

```diff
@@ -0,0 +1,210 @@
+    });
+    _remoteConfigSettings = remoteConfigSettings;
+    return new Future<void>.value();
+  }
+
+  /// Fetches parameter values for your app. Parameter values may be from
+  /// Default Config (local cache) or Remote Config if enough time has elapsed
+  /// since parameter values were last fetched from the server. The default
+  /// expiration time is 12 hours. Expiration must be defined in seconds.
+  Future<void> fetch({Duration expiration: const Duration(hours: 12)}) async {
+    try {
+      final Map<dynamic, dynamic> properties = await _channel.invokeMethod(
+          'RemoteConfig#fetch',
+          <dynamic, dynamic>{'expiration': expiration.inSeconds});
+      _lastFetchTime =
+          new DateTime.fromMillisecondsSinceEpoch(properties[lastFetchTimeKey]);
+      _lastFetchStatus = LastFetchStatus.values[properties[lastFetchStatusKey]];
+    } on PlatformException catch (e) {
+      _lastFetchTime =
+          new DateTime.fromMillisecondsSinceEpoch(e.details[lastFetchTimeKey]);
+      _lastFetchStatus = LastFetchStatus.values[e.details[lastFetchStatusKey]];
+      if (e.code == RemoteConfig.fetchFailedThrottledKey) {
+        print('fetch failed throttled');
+        final int fetchThrottleEnd = e.details['FETCH_THROTTLED_END'];
+        throw new FetchThrottledException._(endTimeInMills: fetchThrottleEnd);
+      } else {
+        print('fetch failed unknown');
+        throw new Exception('Unable to fetch remote config');
+      }
+    }
+    print('fetch succeeded');
+    return new Future<void>.value();
+  }
+
+  /// Activates the fetched config. This makes fetched key-values take effect.
+  ///
+  /// The returned Future contains true if the fetched config is different
+  /// from the currently activated config, it contains false otherwise.
+  Future<bool> activateFetched() async {
+    final Map<dynamic, dynamic> rawParameters =
+        await _channel.invokeMethod('RemoteConfig#activate');
+    final Map<String, RemoteConfigValue> fetchedParameters =
+        _parseRemoteConfigParameters(parameters: rawParameters);
+    final MapEquality<String, RemoteConfigValue> mapEquality =
+        const MapEquality<String, RemoteConfigValue>(
+      keys: const Equality<String>(),
+      values: const Equality<RemoteConfigValue>(),
+    );
+    final bool newConfig = mapEquality.equals(_parameters, fetchedParameters);
+    _parameters = fetchedParameters;
+    notifyListeners();
+    return new Future<bool>.value(newConfig);
+  }
+
+  /// Sets the default config. Default config parameters should be set then when
+  /// changes are needed the parameters should be updated in the Firebase
+  /// console.
+  Future<void> setDefaults(Map<String, dynamic> defaults) async {
+    await _channel.invokeMethod(
+        'RemoteConfig#setDefaults', <String, dynamic>{'defaults': defaults});
+    // Make defaults available even if fetch fails.
+    defaults.forEach((String key, dynamic value) {
```

Comment:
Is there a reason why we are keeping around the old values and aren't just replacing `defaults` with the user-provided `Map`

Reply:
1. This allows defaults to be available without the need to call activate.
1. There may be values in a fetched config that are not in the defaults.

One option to clean this up is to separate fetched parameters from default parameters. So there is no need to merge defaults with existing parameters.